tests/e2e_tests/integration/07-saml.spec.ts

```diff
@@ -0,0 +1,183 @@
+      await page.waitForSelector('text=Entity ID');
+      console.log('waiting over continuing.');
+
+      const response = await axios({
+        url: `${baseUrl}api/management/v1/useradm/sso/idp/metadata`,
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${jwt}`,
+          'Content-Type': 'application/json'
+        },
+        httpsAgent: new https.Agent({
+          rejectUnauthorized: false
+        })
+      });
+
+      if (response.status !== 200) {
+        throw 'expected 201';
+      }
+
+      const id=response.data[0]['id'];
+      console.log(`got metadata id: ${id}`)
+
+      const expected_login_url=`${baseUrl}api/management/v1/useradm/auth/sso/${id}/login`;
+      const find_results = await page.locator('span',{'hasText' : expected_login_url})
+      await expect(find_results).toHaveText(expected_login_url);
+      console.log(`got span with login url: ${await find_results.count()}`)
+
+      const expected_acs_url=`${baseUrl}api/management/v1/useradm/auth/sso/${id}/acs`;
+      const find_results_acs = await page.locator('span',{'hasText' : expected_acs_url})
+      await expect(find_results_acs).toHaveText(expected_acs_url);
+      console.log(`got span with acs url: ${await find_results_acs.count()}`)
+
+      const expected_sp_meta_url=`${baseUrl}api/management/v1/useradm/sso/sp/metadata/${id}`;
+      const find_results_sp_meta = await page.locator('span',{'hasText' : expected_sp_meta_url})
+      await expect(find_results_sp_meta).toHaveText(expected_sp_meta_url);
+      console.log(`got span with SP metadata url: ${await find_results_sp_meta.count()}`)
+
+      const sp_meta_response = await axios({
+        url: `${baseUrl}api/management/v1/useradm/sso/sp/metadata/${id}`,
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${jwt}`,
+          'Content-Type': 'application/json'
+        },
+        httpsAgent: new https.Agent({
+          rejectUnauthorized: false
+        })
+      });
+
+      if (sp_meta_response.status !== 200) {
+        throw 'expected 201';
+      }
+
+      const service_provider_metadata=sp_meta_response.data.replaceAll('Signed="true"','Signed="false"')
+      fs.writeFileSync('fixtures/service_provider_metadata.xml', service_provider_metadata);
+
+      await page.goto('https://samltest.id/upload.php');
+      await page.waitForSelector('input[id=uploader]');
+
+      // Click input[name="userfile"]
+      await page.locator('input[name="userfile"]').click();
+      // Upload m
+      await page.locator('input[name="userfile"]').setInputFiles('fixtures/service_provider_metadata.xml');
+      // Click input:has-text("Upload")
+      await page.locator('input:has-text("Upload")').click();
+      await expect(page).toHaveURL('https://samltest.id/upload.php');
+
+      console.log('uploaded file, making screen shot, after waiting 4s');
+      await page.waitForTimeout(timeout_4s);
+      await page.screenshot({path: 'saml-uploaded.png'});
```

Comment:
this & the other screenshots look like they were used for debug purposes - do you think we need them on regular runs (mind you: we get a video on every first retry...)?

Reply:
My point is, that on case of a successful run we won't care about the screenshots and in case of a failure we have a video that likely contains the image... But feel free to keep them if they help you!

Reply:
thanks :) they really do help me :) I just like a good pic :)